src/api/app/models/project/key_info.rb

```diff
@@ -0,0 +1,34 @@
+class Project
+  class KeyInfo
+    include ActiveModel::Model
+
+    attr_accessor :pubkey, :algorithm, :ssl_certificate
+
+    CACHE_EXPIRY_TIME = 5.minutes
+
+    def self.find_by_project_name(project_name)
+      cache_key = "key_info_project_#{project_name}"
```

Comment:
Yep, exactly! The cache_key gets updated every time you update the project (timestamp). We had in the past some issues with not invalidated cache keys... 
(For this particular case maybe not a problem but just to be consistent)

And maybe you can even just use it in the fetch without the variable:

```
Rails.cache.fetch("key_info_project_#{project.cache_key}" expires_in: CACHE_EXPIRY_TIME)
```